library_generation/library_generation.sh

```diff
@@ -47,9 +47,13 @@ else
 fi
 
 ######## Fix include paths  ########
-INCLUDE_ROS2_PACKAGES=( rmw rcl rcl_action rcl_lifecycle rcl_logging_interface rosidl_runtime_c rosidl_typesupport_interface)
-for var in "${INCLUDE_ROS2_PACKAGES[@]}"; do
-    mv $BASE_PATH/libmicroros/include/${var}/${var}/* $BASE_PATH/libmicroros/include/${var}
+pushd $BASE_PATH/libmicroros/micro_ros_src > /dev/null
+    INCLUDE_ROS2_PACKAGES=$(colcon list | awk '{print $1}' | awk -v d=" " '{s=(NR==1?s:s d)$0}END{print s}')
+popd > /dev/null
+
+apt -y install rsync
+for var in ${INCLUDE_ROS2_PACKAGES}; do
+    rsync -r $BASE_PATH/libmicroros/include/${var}/${var}/* $BASE_PATH/libmicroros/include/${var}
     rm -rf $BASE_PATH/libmicroros/include/${var}/${var}
```

Comment:
```suggestion
    if [ -d "$BASE_PATH/libmicroros/include/${var}/${var}" ]; then
        rsync -r $BASE_PATH/libmicroros/include/${var}/${var}/* $BASE_PATH/libmicroros/include/${var}
        rm -rf $BASE_PATH/libmicroros/include/${var}/${var}
    fi
```
Only move if the directory exist